changelog/1830.feature.rst

```diff
@@ -0,0 +1 @@
+A context manager ``does_not_raise`` is added to complement ``raises`` in parametrized tests with conditional raises.
```

Comment:
I thought we agreed that we were **not** going to add this?  It encourages tests which are of bad form

https://github.com/pytest-dev/pytest/issues/1830#issuecomment-425640691

#4324 was a followup to #1830 that we _document_ how to implement `does_not_raises` if you absolutely need it, not that we actually implement it.

I was and still am :-1: on this.